src/test/java/com/jwplayer/southpaw/util/ByteArraySetTest.java

```diff
@@ -46,6 +74,651 @@ public ByteArraySet createSmallSet() {
+            alreadySeenVals.add(val);
+            assertTrue(vals.contains(val));
+            seenCount++;
+        }
+        assertEquals(size, seenCount);
+    }
+
+    @Test
+    public void emptyToArray() {
+        testToArray(SIZE_EMPTY);
+    }
+
+    @Test
+    public void reallySmallToArray() {
+        testToArray(SIZE_REALLY_SMALL);
+    }
+
+    @Test
+    public void smallToArray() {
+        testToArray(SIZE_SMALL);
+    }
+
+    @Test
+    public void regularToArray() {
+        testToArray(SIZE_REGULAR);
+    }
+
+    @Test
+    public void bigToArray() {
+        testToArray(SIZE_BIG);
+    }
+
+    @Test
+    public void reallyBigToArray() {
+        testToArray(SIZE_REALLY_BIG);
+    }
+
+    private void testContains(int size) {
+        System.out.println("Testing contains method with " + size + " size");
+        ByteArraySet set = new ByteArraySet();
+
+        List<ByteArray> vals = getRandomByteArrays(size);
+
+        for (ByteArray val: vals) {
+            assertTrue(set.add(val));
+        }
+        if (size > 0) {
+            assertFalse(set.isEmpty());
+        } else {
+            assertTrue(set.isEmpty());
+        }
+        Collections.shuffle(vals);
+
+        for(ByteArray val: vals) {
+            assertTrue(set.contains(val));
+        }
+
+        List<ByteArray> absentVals = getRandomByteArrays(10);
+        for(ByteArray val: absentVals) {
+            if (vals.contains(val)) {
+                assertTrue(set.contains(val));
+            } else {
+                assertFalse(set.contains(val));
+            }
+        }
+    }
+
+    @Test
+    public void emptyContains() {
+        testContains(SIZE_EMPTY);
+    }
+
+    @Test
+    public void reallySmallContains() {
+        testContains(SIZE_REALLY_SMALL);
+    }
+
+    @Test
+    public void smallContains() {
+        testContains(SIZE_SMALL);
+    }
+
+    @Test
+    public void regularContains() {
+        testContains(SIZE_REGULAR);
+    }
+
+    @Test
+    public void bigContains() {
+        testContains(SIZE_BIG);
+    }
+
+    @Test
+    public void reallyBigContains() {
+        testContains(SIZE_REALLY_BIG);
+    }
+
+    @Test
+    public void testEmptyLastValueChunkIteratorBug() {
+        System.out.println("Testing iterator bug for a chunk with an empty last value");
+        ByteArraySet set = new ByteArraySet();
+
+        /*
+         * This bug only arises when:
+         *  - there are multiple chunks
+         *  - the first chunk has trailing zeros (a.k.a its last value was deleted)
+         */
+
+        int ChunkByteArrayCount = 2 * PER_CHUNK_MAX_BYTE_ARRAY_COUNT;
+        int FrontingSetByteArrayCount = Math.min(5, ByteArraySet.MAX_FRONTING_SET_SIZE - 1);
+        assertTrue(FrontingSetByteArrayCount > 0);
+
+        int size = ChunkByteArrayCount + FrontingSetByteArrayCount;
+
+        Set<ByteArray> vals = new TreeSet<>();
+        for (ByteArray val: getRandomByteArrays(size)) {
+            vals.add(val);
+        }
```

Comment:
```suggestion
vals.addAll(getRandomByteArrays(size));
```
can be simplified with